src/pages/register-page/types.ts

```diff
@@ -1,8 +1,25 @@
-type InputFieldType = 'text' | 'email' | 'password' | 'date' | 'country-select';
+export type FormValues = {
+  email: string;
+  password: string;
+  firstName: string;
+  lastName: string;
+  birth: string;
+  country: string;
+  city: string;
+  street: string;
+  postCode: string;
+};
+
+export type InputFieldType =
+  | 'text'
+  | 'email'
+  | 'password'
+  | 'date'
+  | 'country-select';
 
-export type FormField = {
+export type FormField<Name extends keyof FormValues = keyof FormValues> = {
   type: InputFieldType;
   label: string;
-  name: string;
+  name: Name;
```

Comment:
```suggestion
  name: FieldName;;
```
ты используешь имена как значения. стоит вынести их в енам. не пугайся, это тебе жизнь упростит))